lib/redis_memo/memoizable/dependency.rb

```diff
@@ -12,25 +12,35 @@ def memos
     @nodes.values
   end
 
-  def depends_on(memo_or_model, **conditions)
-    if !memo_or_model.is_a?(RedisMemo::Memoizable)
+  def depends_on(dependency, **conditions)
+    case dependency
+    when self.class
+      nodes.merge!(dependency.nodes)
+    when RedisMemo::Memoizable
+      memo = dependency
+      return if nodes.include?(memo.cache_key)
+      nodes[memo.cache_key] = memo
+
+      if memo.depends_on
+        # Extract dependencies from the current memoizable and recurse
+        instance_exec(&memo.depends_on)
+      end
+    when using_active_record?
       [
-        memo_or_model.redis_memo_class_memoizable,
-        RedisMemo::MemoizeQuery.create_memo(memo_or_model, **conditions),
+        dependency.redis_memo_class_memoizable,
+        RedisMemo::MemoizeQuery.create_memo(dependency, **conditions),
       ].each do |memo|
         nodes[memo.cache_key] = memo
       end
-
-      return
+    else
+      raise(
+        RedisMemo::ArgumentError,
+        "Invalid dependency type #{dependency.class}"
+      )
     end
+  end
 
-    memo = memo_or_model
-    return if nodes.include?(memo.cache_key)
-    nodes[memo.cache_key] = memo
-
-    if memo.depends_on
-      # Extract dependencies from the current memoizable and recurse
-      instance_exec(&memo.depends_on)
-    end
+  def using_active_record?
```

Comment:
For better performance, we can use a static class tha responds to `.call` instead of runtime proc:
```ruby
class UsingActiveRecord
  def self.call
  end
end

Reply:
https://github.com/chanzuckerberg/redis-memo/pull/12